lib/legacy/polymer.dom.html

```diff
@@ -55,9 +55,9 @@
      * Returns an instance of `Polymer.FlattenedNodesObserver` that
      * listens for node changes on this element.
      *
-     * @param {Function} callback Called when direct or distributed children
+     * @param {function(!Element, { target: !Element, addedNodes: !Array<!Element>, removedNodes: !Array<!Element> }):void} callback Called when direct or distributed children
```

Comment:
I would extract this object type to an internal type. You are using it later in `FlattenedNodesObserver` as well, which is a little bit of duplication.

Reply:
Only downside to that is that we'd also have to also manually write it for typescript in our `extra-types.d.ts`, because analyzer doesn't pick up on type declarations.